service/servicehttp/redirectHandler.go

```diff
@@ -50,6 +51,10 @@ func (rh *RedirectHandler) ServeHTTP(response http.ResponseWriter, request *http
 	instance += strings.TrimRight(request.RequestURI, "/")
 	rh.Logger.Log(level.Key(), level.DebugValue(), logging.MessageKey(), "redirecting", "instance", instance)
 
+	logHeaderFunc, ctx := logginghttp.SetLogger(rh.logger, logginghttp.Header("X-WebPA-Device-Name", "device_id"), logginghttp.Header("Authorization", "authorization"))
```

Comment:
But this will not get logged in the error case of 400 in line 39 above. I want this information during error cases only.

Reply:
okay give me a bit